### PR TITLE
fix Delta.DifferentAt behavior via changes to Path.Contains

### PR DESCRIPTION
**Pull request by echen-98**
Fixes issue https://github.com/aws-controllers-k8s/community/issues/808

### CODE REVIEW

File: pkg/compare/path.go [hunk 1]
```diff
@@ -51,14 +51,30 @@ func (p Path) Pop() {
 	}
 }
 
-// Contains returns true if the supplied string appears within the Path
+// TODO: case sensitivity?
```
Comment (echen-98) [inline pkg/compare/path.go:54]: should we keep case-sensitive comparison?

File: pkg/compare/delta_test.go [hunk 2]
```diff
@@ -53,11 +53,14 @@ func TestDifferentAt(t *testing.T) {
...
 	d = compare.NewDelta()
 	d.Add("Baz.Y", a.Baz.Y, b.Baz.Y)
 	require.True(d.DifferentAt("Baz"))
-	require.True(d.DifferentAt("Y"))
-	require.False(d.DifferentAt("Bar"))
+	require.True(d.DifferentAt("Baz.Y"))
+	require.False(d.DifferentAt("Y")) // there is no diff for top-level field "Y"
+	require.False(d.DifferentAt("Bar")) // diff exists but it was not added to Delta
+	require.False(d.DifferentAt("Baz.Y.Z")) // subject length exceeds length of diff Path
+	require.False(d.DifferentAt("Baz.Z"))  // matches Path top-level field but not sub-field
```
Comment (RedbackThomson) [inline pkg/compare/delta_test.go:65]: ++ Super clear test cases! Love it!

File: pkg/compare/path.go (hunk 1 above)
Comment (RedbackThomson) [reply pkg/compare/path.go:54]: I think so, yes. We expect case-sensitivity in all Go fields and I'd expect that these strings are reflections of Go fields.

**Review by RedbackThomson — APPROVED** [review]

File: pkg/compare/path.go (hunk 1 above)
Comment (echen-98) [reply pkg/compare/path.go]: Ok, removed TODO comment

File: pkg/compare/path.go [hunk 3]
```diff
@@ -51,14 +51,29 @@ func (p Path) Pop() {
...
-// Contains returns true if the supplied string appears within the Path
+/* Contains returns true if the supplied string, delimited on ".", matches
+	p.parts up to the length of the supplied string.
+	e.g. if the Path p represents "A.B":
+		subject "A" -> true
+		subject "A.B" -> true
+		subject "A.B.C" -> false
+		subject "B" -> false
+		subject "A.C" -> false
+ */
```
[the remark this answers is not in the document]
Comment (A-Hilaly) [reply pkg/compare/path.go:62]: You can also instruct `godoc` to generate testable examples in the package documentation: https://blog.golang.org/examples / https://blog.golang.org/godoc

**Review by A-Hilaly — COMMENTED** [review]
Very nice, thanks Eric!
I left one comment below

File: pkg/compare/delta_test.go (hunk 2 above)
Comment (jaypipes) [reply pkg/compare/delta_test.go:65]: agreed, excellent clarity @echen-98

File: pkg/compare/path.go [hunk 4]
```diff
@@ -51,14 +51,28 @@ func (p Path) Pop() {
 	}
 }
 
-// Contains returns true if the supplied string appears within the Path
+//	Contains returns true if the supplied string, delimited on ".", matches
+//	p.parts up to the length of the supplied string.
```
Comment (jaypipes) [inline pkg/compare/path.go:55]: femto-nit: not sure if it's your IDE or editor changing the above, but comments should have a single space (not `\t`) after the `//` and before the first letter of the comment...

**Review by jaypipes — APPROVED** [review]
Great stuff @echen-98 :) Thank you!

File: pkg/compare/path.go (hunk 4 above)
Comment (echen-98) [reply pkg/compare/path.go:55]: Got it, will keep in mind for next time

**Pull request merged**